packages/wonder-blocks-cell/src/components/internal/cell-core.tsx

```diff
@@ -250,43 +250,76 @@ const styles = StyleSheet.create({
     /**
      * States
      */
-    hovered: {
-        background: Color.offBlack8,
-    },
+    clickable: {
+        outline: "none",
+        /**
+         * States
+         */
+        // disabled
+        // NOTE: We use `aria-disabled` instead of `disabled` because we want
+        // to allow the cell to be focusable even when it's disabled.
+        [":hover[aria-disabled=true]" as any]: {
+            cursor: "not-allowed",
+        },
 
-    // Handling the focus ring internally because clickable doesn't support
-    // rounded focus ring.
-    focused: {
-        borderRadius: Spacing.xxxSmall_4,
-        outline: `solid ${Spacing.xxxxSmall_2}px ${Color.blue}`,
-        // The focus ring is not visible when there are stacked cells.
-        // Using outlineOffset to display the focus ring inside the cell.
-        outlineOffset: -Spacing.xxxxSmall_2,
-        // To hide the internal corners of the cell.
-        overflow: "hidden",
-    },
+        // focus (only visible when using keyboard navigation)
+        ":focus-visible": {
+            borderRadius: Spacing.xxxSmall_4,
+            // To hide the internal corners of the cell.
+            overflow: "hidden",
+            // To display the focus ring based on the cell's border.
+            position: "relative",
+        },
+        // NOTE: We use a pseudo element to draw the focus ring because we can't
+        // use `outline` since it conflicts with different layout contexts (e.g.
+        // `View` elements add their own z-index).
+        [":focus-visible:after" as any]: {
+            content: "''",
+            // Since we are using a pseudo element, we need to manually
+            // calculate the width/height and use absolute position to
+            // prevent other elements from being shifted around.
+            position: "absolute",
+            top: 0,
+            left: 0,
+            zIndex: 1,
+            // We remove the border width from the width/height to ensure
+            // that the focus ring is drawn inside the cell.
+            width: `calc(100% - ${Spacing.xxxSmall_4}px)`,
+            height: `calc(100% - ${Spacing.xxxSmall_4}px)`,
```

Comment:
**Praise:** oooo I see. It's unfortunate that `calc` is necessary, but I get why it has to be done this way. I'll probably end up having to use this myself soon too. Nice solve!